platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DefaultConfigurationCache.kt

```diff
@@ -596,6 +600,8 @@ class DefaultConfigurationCache internal constructor(
         cacheIO.withReadContextFor(fingerprintFile.stateType, fingerprintFile::inputStream) { codecs ->
             withIsolate(IsolateOwners.OwnerHost(host), codecs.fingerprintTypesCodec()) {
                 action(object : ConfigurationCacheFingerprintController.Host {
+                    override val buildPath: Path
+                        get() = host.service<GradleInternal>().identityPath
```

Comment:
🌈  I wonder if there is already many places where we reach for the build path like this. If so, it might be time to introduce a simpler service that just provides the needed information, rather than reaching out to the hairy `GradleInternal`